docs/book/best-practice-identity.md

```diff
@@ -0,0 +1,123 @@
+
+### Flow
+Two things happen independently with this kind of LDAP/AD integration.
+* (1) OPA downloads new policy bundles including LDAP/AD
+* (A) OPA-enabled software system asks OPA for policy decisions
+
+![Bundle flow](images/best-practice-identity-bundle.png)
+
+### Updates
+The lag between an LDAP/AD update and OPA having the update is the sum of the lag for an update between LDAP/AD and the central server and the lag for an update between the central server an OPA.  So if LDAP/AD updates every 5 minutes, and OPA pulls an update every 2 minutes, then the total maximum lag is 7 minutes.  Unlike the JWT case, it is feasible that a user could perform an action requiring authorization on the OPA-enabled service before OPA has the appropriate LDAP/AD policy, but you can account for that when writing policy and reject any request when there is insufficient data.
+
+### Size limitations
+Unlike the JWT case, where OPA only stores 1 user's LDAP/AD data at a time, when synchronizing LDAP/AD, OPA stores all users' LDAP/AD data at once in memory.  Obviously this can be a problem with large LDAP/AD stores.  Because the centralized server handles both policy and data it could provide just the LDAP/AD data necessary for the policies.
+
+<!-- **Security**
+* Don't expose OPA's API except through localhost
+* Assuming LDAP/AD is the only context needed, use OPA's authentication/authorization to disable all of OPA's APIs except those needed by the OPA-enabled service. -->
+
+## 3. Pushing LDAP/AD into OPA
+
+Another way to replicate LDAP/AD data into OPA is to use OPA's API for injecting arbitrary JSON data.  You can build a synchronizer that pulls information out of LDAP/AD and pushes that information in OPA through its API.
+
+### Flow
+
+Two things happen independently with this kind of LDAP/AD integration.
+* (1) Synchronizer keeps OPA up to date with LDAP/AD
+* (A) OPA-enabled software system asks OPA for policy decisions
+
+![Push flow](images/best-practice-identity-push.png)
+
+### Updates
+
+The total lag between LDAP/AD being updated and OPA being updated is the sum of the lag for an update between LDAP/AD and the synchronizer plus the lag for an update between the synchronizer and OPA.  Unlike the JWT case, it is feasible that a user could perform an action requiring authorization on the OPA-enabled service before OPA has the appropriate LDAP/AD policy, but you can account for that when writing policy and reject any request when there is insufficient data.
+
+### Size limitations
+Unlike the JWT case, where OPA only stores 1 user's LDAP/AD data at a time, when synchronizing LDAP/AD, OPA stores all users' LDAP/AD data at once in memory.  Obviously this can be a problem with large LDAP/AD stores.
+
+<!--
+**Security**
```

Comment:
Are these sections commented out on purpose?

Reply:
Yes I left them commented out because they seemed to require more thought than I had originally expected.  But I made some progress.  Can remove if you'd prefer.